### PR TITLE
Make FindBin strict and warnings compliant

### PR DESCRIPTION
**Pull request by toddr**

### CODE REVIEW

File: dist/FindBin/lib/FindBin.pm [hunk 1]
```diff
@@ -107,8 +109,8 @@ sub cwd2 {
 
 sub init
 {
- *Dir = \$Bin;
- *RealDir = \$RealBin;
+ *Dir = *Bin;
```
Comment (Grinnz) [inline dist/FindBin/lib/FindBin.pm]: What's the reason for this change?

Comment (Grinnz) [reply dist/FindBin/lib/FindBin.pm]: For clarity since glob assignment is black magic stuff, `*foo = \$bar` aliases the SCALAR slot of `*bar` to the SCALAR slot of `*foo`, whereas `*foo = *bar` aliases *all* slots - ARRAY, HASH, CODE, IO, etc. Only the scalar slot is used by this module so the existing code should be fine.

Comment (toddr) [reply dist/FindBin/lib/FindBin.pm]: You're suggesting it would be cheaper as an RV. Probably true.

Comment (karenetheridge) [reply dist/FindBin/lib/FindBin.pm]: This is an alias, not a reference.
